Fix flaky CI: `wrapper-validation` fails with `Error: connect ITIMEDOUT`

<img width="1392" alt="Image" src="https://github.com/user-attachments/assets/01feaf27-624b-4fa1-a401-be46ce10fd53" />

https://github.com/nicokosi/java-for-katas/actions/runs/14964218251/job/42031212891

```
025-05-12T05:03:51.0392576Z ##[group]Run gradle/actions/wrapper-validation@v3.5.0
2025-05-12T05:03:51.0393033Z with:
2025-05-12T05:03:51.0393203Z   min-wrapper-count: 1
2025-05-12T05:03:51.0393407Z   allow-snapshots: false
2025-05-12T05:03:51.0393595Z env:
2025-05-12T05:03:51.0393830Z   JAVA_HOME: /opt/hostedtoolcache/Java_Zulu_jdk/24.0.1-9/x64
2025-05-12T05:03:51.0394229Z   JAVA_HOME_24_X64: /opt/hostedtoolcache/Java_Zulu_jdk/24.0.1-9/x64
2025-05-12T05:03:51.0394617Z   GRADLE_ACTION_ID: gradle/wrapper-validation-action
2025-05-12T05:03:51.0394891Z ##[endgroup]
2025-05-12T05:03:52.5958195Z ##[error]Multiple errors returned
2025-05-12T05:03:52.5960090Z ##[error]Error 0: connect ETIMEDOUT 104.16.72.101:443
2025-05-12T05:03:52.5960654Z Error: connect ETIMEDOUT 104.16.72.101:443
2025-05-12T05:03:52.5960968Z     at createConnectionError (node:net:1652:14)
2025-05-12T05:03:52.5961331Z     at Timeout.internalConnectMultipleTimeout (node:net:1711:38)
2025-05-12T05:03:52.5961699Z     at listOnTimeout (node:internal/timers:583:11)
2025-05-12T05:03:52.5962055Z     at process.processTimers (node:internal/timers:519:7)
2025-05-12T05:03:52.5963644Z ##[error]Error 1: connect ENETUNREACH 2606:4700::6810:4865:443 - Local (:::0)
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -14,6 +14,6 @@ jobs:
           java-version: '23'
           distribution: 'zulu'
       - name: Validate Gradle wrapper
-        uses: gradle/wrapper-validation-action@f9c9c575b8b21b6485636a91ffecd10e558c62f6
+        uses: gradle/actions/wrapper-validation@v4
       - name: Build with Gradle
         run: ./gradlew build

